src/libfsm/walk2.c

```diff
@@ -250,7 +250,8 @@ fsm_walk2_tuple_new(struct fsm_walk2_data *data,
 	if (comb == NULL) {
 		return NULL;
 	}
-	comb->equiv = NULL;
+
+	assert(comb->tmp.visited == NULL);
```

Comment:
Why would this be true? As a side effect of setting another field in the union?

Reply:
Oh because `fsm_state_clear_tmp()` memsets the union to `0x0`. Okay. I'll merge this, but I'd prefer to come back and do that per field in the union instead (and the same for `memset()` elsewhere).